src/toil/batchSystems/registry.py

```diff
@@ -71,3 +71,10 @@ def kubernetes_batch_system_factory():
 }
 BATCH_SYSTEMS = list(BATCH_SYSTEM_FACTORY_REGISTRY.keys())
 DEFAULT_BATCH_SYSTEM = 'single_machine'
+
+"""
+Adds a batch system to the registry for workflow-supplied batch systems.
+"""
+def addBatchSystemFactory(key, batchSystemFactory):
```

Comment:
The docstring is supposed to be inside the function, not before it.

Also the function signature needs type hints. I think batchSystemFactory is maybe a `Callable[[], Type[AbstractBatchSystem]]`.